eclipse-collections/src/main/java/org/eclipse/collections/impl/list/IntervalUtils.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019 Goldman Sachs and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.list;
+
+public final class IntervalUtils
+{
+    private IntervalUtils()
+    {
+        throw new AssertionError("Suppress default constructor for noninstantiability");
+    }
+
+    public static void throwOnIllegalArguments(long from, long to, long stepBy)
+    {
+        IntervalUtils.throwOnIllegalStepBy(from, to, stepBy);
+        IntervalUtils.throwOnIllegalSize(from, to, stepBy);
+    }
+
+    private static void throwOnIllegalSize(long from, long to, long stepBy)
+    {
+        long rangeSize = IntervalUtils.rangeSize(from, to, stepBy);
+        if ((long) Integer.MAX_VALUE < rangeSize)
+        {
+            throw new IllegalArgumentException("Range size: "
+                    + rangeSize
+                    + " exceeds max size() of "
+                    + Integer.MAX_VALUE);
+        }
+    }
+
+    private static void throwOnIllegalStepBy(long from, long to, long stepBy)
+    {
+        if (stepBy == 0)
+        {
+            throw new IllegalArgumentException("Cannot use a step by of 0");
+        }
+        if (from > to && stepBy > 0 || from < to && stepBy < 0)
+        {
+            throw new IllegalArgumentException("Step by is incorrect for the range");
+        }
+        // Original Version
+        // if ((from > to && (from + stepBy < to)) || (from < to && (from + stepBy > to)))
+        if ((from > to && (stepBy < to - from)) || (from < to && (stepBy > to - from)))
```

Comment:
this will be a breaking change no? Currently this test passes:

```
    @Test
    public void interval()
    {
        Interval integers = Interval.fromToBy(0, 2, 3);
        Assert.assertEquals(Lists.mutable.with(0), integers.toList());
    }

```

Reply:
You are correct.